src/pages/HomePage.jsx

```diff
@@ -0,0 +1,139 @@
+import { useState, useEffect, useMemo } from "react";
+import { Navigate } from "react-router-dom";
+import Modal from "../Components/Modal/Modal";
+import NoticeModal from "../Components/Modal/NoticeModal";
+import DoneModal from "../Components/Modal/DoneModal";
+import TodoListItem from "../Components/TodoList/TodoListItem";
+import ProgressBar from "../Components/ProgressBar/ProgressBar";
+import Button from "../Components/Button/Button";
+import "../App.css";
+import { firestore } from "../service/firebase";
+import {
+  collection,
+  query,
+  where,
+  updateDoc,
+  getDocs,
+  deleteDoc,
+} from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+import styles from "./HomePage.module.css";
+import { AnimatePresence } from "framer-motion";
+
+const HomePage = ({ userInfo, isLoggedIn }) => {
+  const [showModal, setShowModal] = useState(false);
+  const [todoItems, setTodoItems] = useState([]);
+  const queryPath = `/users/${userInfo?.uid}/todos`;
+  const _query = collection(firestore, queryPath);
+  const [todos, loading, error] = useCollectionData(_query);
+  const [editTarget, setEditTarget] = useState();
+  const [showDone, setShowDone] = useState(false);
+
+  const progressPercentage = useMemo(() => {
+    let ret = 0;
+    if (todoItems) {
+      if (todoItems.length > 0)
+        ret = todoItems.filter((el) => el.isComplete).length / todoItems.length;
+      else ret = 0;
+    }
```

Comment:
중첩 if문은 피할수 있으면 피하는게 좋습니다.
```suggestion
    if(!todlItems || todoItems.length === 0) return 0
    return todoItems.filter((el) => el.isComplete).length / todoItems.length;
```

Reply:
안그래도 이거 때문에 찝찝했는데 시원한 해결 제시 감사합니다